mediapipe/tasks/ios/genai/inference/sources/LlmInference.swift

```diff
@@ -100,123 +80,49 @@ import MediaPipeTasksGenAIC
   }
 
-  /// Generates a response based on the input text asynchronously. The `progress` callback returns
-  /// the partial responses from the LLM or any errors. `completion` callback is invoked once the
-  /// LLM is done generating responses.
-  ///
-  /// - Parameters:
-  ///   - progress: A callback invoked when a partial response is available from the LLM.
-  ///   - completion: A callback invoked when the LLM finishes response generation.
-  /// - Throws: An error if the LLM's response is invalid.
-  @objc public func generateResponseAsync(
-    inputText: String,
-    progress: @escaping (_ partialResponse: String?, _ error: Error?) -> Void,
-    completion: @escaping (() -> Void)
-  ) throws {
-    /// Disallow response generation if another response generation call is already in progress.
-    try shouldContinueWithResponseGeneration()
-
-    /// Used to make a decision about whitespace stripping.
-    var receivedFirstToken = true
-
-    try llmTaskRunner.predict(
-      inputText: inputText,
-      progress: { partialResponseStrings, error in
-
-        guard let responseStrings = partialResponseStrings,
-          let humanReadableLlmResponse = LlmInference.humanReadableString(
-            llmResponses: responseStrings, stripLeadingWhitespaces: receivedFirstToken)
-        else {
-          progress(nil, GenAiInferenceError.invalidResponse)
-          return
-        }
-
-        /// Reset state after first response is processed.
-        receivedFirstToken = false
-
-        progress(humanReadableLlmResponse, nil)
-      },
-      completion: { [weak self] in
-        self?.responseGenerationInProgress = false
-        completion()
-      })
-  }
-
-  /// Generates a response based on the input text asynchronously.
-  ///
-  /// - Parameters:
-  ///   - inputText: The prompt used to query the LLM.
-  /// - Returns: An async throwing stream that contains the partial responses from the LLM.
-  @available(iOS 13, macOS 10.15, tvOS 13, watchOS 6, *)
-  public func generateResponseAsync(inputText: String) -> AsyncThrowingStream<String, Error> {
-    AsyncThrowingStream { continuation in
-      do {
-        try generateResponseAsync(
-          inputText: inputText,
-          progress: { partialResponse, error in
-            if let error {
-              continuation.finish(throwing: error)
-            } else if let partialResponse {
-              continuation.yield(partialResponse)
-            }
-          },
-          completion: {
-            continuation.finish()
-          })
-      } catch {
-        continuation.finish(throwing: error)
+  /// If no response generation using this `llmInference` is currently in progress, this function
+  /// updates the response generation state and returns successfully granting access to its caller
+  /// to execute response generation. If response generation is already in progress, throws an
+  /// error.
+  /// Any
+  func shouldContinueWithResponseGeneration() throws {
```

Comment:
@yishuangP I am maintaining a response generation state in the inference which the session can read and update through these methods to prevent simultaneous response generation calls from multiple sessions.